src/main/java/com/ldtteam/blockui/Render.java

```diff
@@ -0,0 +1,79 @@
+package com.ldtteam.blockui;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.mojang.math.Quaternion;
+import com.mojang.math.Vector3f;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.client.renderer.entity.EntityRenderDispatcher;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.LivingEntity;
+
+/**
+ * Render utility functions.
+ */
+public final class Render

```

Comment:
UiRenderMacros class

Reply:
I put it here because this is where it was in 1.16... and it seems weird to have this sort of code in a base class for all controls, when it's not going to be useful for any others.

Also, in a standalone class it could potentially be called directly by Minecolonies when it wants to do this sort of rendering not as part of an EntityIcon, to avoid duplicating the rendering code.  That makes less sense if it's part of UiRenderMacros.

Reply:
UiRenderMacros was created to have unified place for rendering methods, Pane extends UiRenderMacros so you can call this whenever you have any gui element

Reply:
Having it in a single class makes some sense.  Having that be the base class makes no sense whatsoever, it's an abuse of inheritance.

But fine, whatever.